### PR TITLE
Format everything, add gotests Makefile

### PR DESCRIPTION
**Pull request by jsravn**
- Run gofmt/lint/vet, do this automatically in Travis as well.
- Created a gotests specific Makefile so we can just work in that
directory. As a result I had to rejig the other Makefile/travis.yml.

### CODE REVIEW

File: enterprise-suite/gotests/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+vet:
+	go vet ./...
+
+.PHONY: lint
+lint:
+	golint -set_exit_status $(go list ./...)
+
+.PHONY: go-tests
+run-tests:
+	ginkgo -r -- --namespace=$(TILLER_NAMESPACE) --tiller-namespace=$(TILLER_NAMESPACE)
```
Comment (dbrinegar) [inline enterprise-suite/gotests/Makefile]: just double checking that TILLER_NAMESPACE is intended to be used for two namespace things here..

Comment (jsravn) [reply enterprise-suite/gotests/Makefile]: It's a little confusing - it should install in the same namespace both things. I'll change it...

File: enterprise-suite/gotests/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+.PHONY: go-tests
+run-tests:
+	ginkgo -r -- --namespace=$(NAMESPACE) --tiller-namespace=$(NAMESPACE)
+
+.PHONY: format
+format:
+	goimports -w $(files)
+
+.PHONY: setup-tools
+setup-tools:
```
Comment (srikanthops) [inline enterprise-suite/gotests/Makefile]: can we call setup-tools as part of check

Comment (jsravn) [reply enterprise-suite/gotests/Makefile]: setup-tools is quite slow - so I didn't want to do this every time.

**Pull request merged**